Applications/mnist/Tensorflow/Training_Keras.py

```diff
@@ -0,0 +1,183 @@
+
+            yield x_batch, y_batch
+            
+##
+# @brief load input data from file
+# @return (InputVector, InputLabel, Validation Vector, ValidationLabel)
+def load_data():
+    tf.compat.v1.set_random_seed(777)
+
+    data_size = TOTAL_TRAIN_DATA_SIZE;
+    InputVector = np.zeros((TOTAL_LABEL_SIZE*data_size, FEATURE_SIZE),dtype=np.float32)
+    InputLabel = np.zeros((TOTAL_LABEL_SIZE*data_size, TOTAL_LABEL_SIZE),dtype=np.float32)
+
+    ValVector = np.zeros((TOTAL_LABEL_SIZE*TOTAL_VAL_DATA_SIZE,FEATURE_SIZE),dtype=np.float32)
+    ValLabel = np.zeros((TOTAL_LABEL_SIZE*TOTAL_VAL_DATA_SIZE, TOTAL_LABEL_SIZE),dtype=np.float32)
+
+    #read Input & Label    
+
+    fin = open('mnist_trainingSet.dat','rb')
+    for i in range(TOTAL_LABEL_SIZE*TOTAL_VAL_DATA_SIZE):
+        for j in range(FEATURE_SIZE):
+            data_str = fin.read(4)
+            ValVector[i,j] = struct.unpack('f',data_str)[0]
+        for j in range(TOTAL_LABEL_SIZE):
+            data_str = fin.read(4)
+            ValLabel[i,j] = struct.unpack('f',data_str)[0]
+    fin.close()
+
+    fin=open('mnist_trainingSet.dat','rb')
+    for i in range(TOTAL_LABEL_SIZE*data_size):
+        for j in range(FEATURE_SIZE):
+            data_str = fin.read(4)
+            InputVector[i,j] = struct.unpack('f',data_str)[0]
+            
+        for j in range(TOTAL_LABEL_SIZE):
+            data_str = fin.read(4)
+            InputLabel[i,j] = struct.unpack('f',data_str)[0]
+    fin.close()
+
+    return InputVector, InputLabel, ValVector, ValLabel
+
+##
+# @brief training loop
+#        - epoches : 1500
+#        - Optimizer : Adam
+#        - Activation : softmax
+#        - loss : cross entropy
+#
+def train_nntrainer():
+    InVec, InLabel, ValVec, ValLabel = load_data()    
+
+    print('reading is done')
+    inputs = tf.placeholder(tf.float32, [None, FEATURE_SIZE], name="input_X")
+    labels = tf.placeholder(tf.float32,[None, TOTAL_LABEL_SIZE], name = "label")
+
+    model = models.Sequential()
+    model.add(Conv2D(6, (5,5), padding='valid', activation='sigmoid', input_shape=(28,28,1)))
+    model.add(AveragePooling2D(pool_size=(2,2)))
+    model.add(Conv2D(12, (5,5), padding='valid', activation='sigmoid'))
+    model.add(AveragePooling2D(pool_size=(2,2)))
+    model.add(Flatten())
+    model.add(layers.Dense(10,activation='softmax',))
+
+    model.compile(optimizer = optimizers.Adam(lr=1e-4),
+                  loss=tf.keras.losses.CategoricalCrossentropy(),
+                  metrics=['accuracy'])
+
+    model.summary()
+
+    model_Hist = History_LAW()
+
+    history = model.fit(datagen(InVec, InLabel, batch_size), epochs=num_epoch, steps_per_epoch=len(InVec)//batch_size, validation_data=datagen(ValVec, ValLabel, batch_size), validation_steps=len(ValVec)//batch_size, callbacks=[model_Hist])
+    
+    count =0
+    
+    if not os.path.exists('./nntrainer_tfmodel'):
+        os.mkdir('./nntrainer_tfmodel')
+        model.save('./nntrainer_tfmodel/nntrainer_keras.h5')
```

Comment:
Model is not saved when `os.path.exists('nntrainer_tfmodel')`
Maybe this?
```suggestion
    model.save('./nntrainer_tfmodel/nntrainer_keras.h5')
```

Reply:
done. Thanks